helpers/object_utils.ts

```diff
@@ -3,12 +3,9 @@ export function deepFreeze<T extends {}>(obj: T): Readonly<T> {
 
   for (const k of Object.keys(obj)) {
     const key = k as keyof T;
-    if (
-      obj[key] !== null &&
-      (typeof obj[key] === 'object' || typeof obj[key] === 'function') &&
-      !Object.isFrozen(obj[key])
-    ) {
-      deepFreeze(obj[key]);
+    const value = obj[key];
```

Comment:
Seems like somehow this PR caused the typechecker to stop believing that `typeof obj[key] === 'object` implies it's still and object on the next line. Solved by extracting a local variable for `obj[key]` to persuade the typechecker that its type is consistent.